src/ripple/app/tx/impl/apply.cpp

```diff
@@ -47,7 +47,12 @@ checkValidity(HashRouter& router,
     if (!(flags & SF_SIGGOOD))
     {
         // Don't know signature state. Check it.
-        auto const sigVerify = tx.checkSign();
+        STTx::RequireFullyCanonicalSig const requireCanonicalSig =
+            rules.enabled(featureRequireFullyCanonicalSig) ?
+            STTx::RequireFullyCanonicalSig::yes :
+            STTx::RequireFullyCanonicalSig::no;
+
+        auto const sigVerify = tx.checkSign(requireCanonicalSig);
```

Comment:
I'd prefer: `auto const requireCanonicalSig = ... whatever ...` instead of `STTx::RequireFullyCanonicalSig const requireCanonicalSig = ... whatever ...`.

Rationale: the precise type of the variable here is irrelevant and doesn't add much. This is, ultimately, a `bool` with some type-safety and user-friendliness sprinked on it, and a programmer can deduce the actual type by examining the ternary.